bundle/direct/dresources/secret_scope.go

```diff
@@ -0,0 +1,92 @@
+package dresources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/bundle/deployplan"
+	"github.com/databricks/cli/libs/utils"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/workspace"
+)
+
+type ResourceSecretScope struct {
+	client *databricks.WorkspaceClient
+}
+
+type SecretScopeConfig struct {
+	workspace.CreateScope
+}
+
+func (*ResourceSecretScope) New(client *databricks.WorkspaceClient) *ResourceSecretScope {
+	return &ResourceSecretScope{
+		client: client,
+	}
+}
+
+func (*ResourceSecretScope) PrepareState(input *resources.SecretScope) *SecretScopeConfig {
+	return &SecretScopeConfig{
+		CreateScope: workspace.CreateScope{
+			Scope:                  input.Name,
+			ScopeBackendType:       input.BackendType,
+			BackendAzureKeyvault:   input.KeyvaultMetadata,
+			InitialManagePrincipal: "",
+			ForceSendFields:        nil,
+		},
+	}
+}
+
+func (*ResourceSecretScope) RemapState(remote *workspace.SecretScope) *SecretScopeConfig {
+	return &SecretScopeConfig{
+		CreateScope: workspace.CreateScope{
+			Scope:                  remote.Name,
+			ScopeBackendType:       remote.BackendType,
+			BackendAzureKeyvault:   remote.KeyvaultMetadata,
+			InitialManagePrincipal: "",
+			ForceSendFields:        utils.FilterFields[workspace.CreateScope](remote.ForceSendFields),
+		},
+	}
+}
+
+func (r *ResourceSecretScope) DoRefresh(ctx context.Context, id string) (*workspace.SecretScope, error) {
+	scopes, err := r.client.Secrets.ListScopesAll(ctx)
```

Comment:
I recall that there is no "get by name" for scopes. Please include a comment stating that this is the case, or readers will be confused why we don't just get the secret scope by name.